### PR TITLE
GPT4All Wrapper

### PR DESCRIPTION
**Pull request by hinthornw**
Adds a wrapper for [pyllamacpp](https://github.com/nomic-ai/pyllamacpp) models.

Tested locally.

Started drafting things for async/streaming on a separate branch but won't get to it today: https://github.com/hinthornw/lckg/tree/william/gpt4all_asyn

### CODE REVIEW

File: docs/modules/models/llms/integrations/gpt4all.ipynb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{
```
Comment (hwchase17) [inline docs/modules/models/llms/integrations/gpt4all.ipynb:1]: you need a markdown title `# GPT4All` as the first cell in this notebook for it to properly render in the documentation

File: docs/ecosystem/gpt4all.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Example:
+
+```python
+model = GPT4All(model="./models/gpt4all-model.bin", n_predict=55, temp=0)
+response = model("Once upon a time, ")
+```
+## Model File
+
+You can find links to model file downloads at the [GPT4all](https://github.com/nomic-ai/gpt4all) repository. They will need to be converted to `ggml` format to work, as specified in the [pyllamacpp](https://github.com/nomic-ai/pyllamacpp) repository.
```
Comment (AndriyMulyar) [inline docs/ecosystem/gpt4all.md:35]: We are going to make this much more smooth very soon (have a ggML compliant model being distributed). I made a note to PR this conversion logic out of the langchain docs but its not a blocker to merge now as this documentation is accurate.

**Review by AndriyMulyar — APPROVED** [review]
This overall looks good. I will be PR'ing updates to this as the ecosystem develops to keep langchain on the bleeding edge :)

File: docs/ecosystem/gpt4all.md (hunk 2 above)
Comment (hinthornw) [reply docs/ecosystem/gpt4all.md:35]: Sounds great. Feel free to open a PR fixing this once the model is available.

**Pull request merged**